package/MDAnalysis/topology/MOL2Parser.py

```diff
@@ -146,17 +158,30 @@ def parse(self, **kwargs):
             resids.append(resid)
             resnames.append(resname)
             charges.append(charge)
+            elements.append(atom_type.strip(".")[0])
 
         n_atoms = len(ids)
 
-        masses = guessers.guess_masses(types)
+        validated_elements = []
+        for elem in elements:
+            if elem.capitalize() in SYMB2Z:
+                validated_elements.append(elem.capitalize())
+            else:
+                wmsg = (f"Unknown element {elem.capitalize()} found for some "
```

Comment:
Putting the warning inside the loop will spam the user if they've got many weird types. Probably a better pattern is to accumulate weird types into a set and if any print after the loop.

Reply:
Good point, I like this very much.